build.rs

```diff
@@ -547,9 +547,13 @@ fn obj_path(out_dir: &Path, src: &Path) -> PathBuf {
 
 fn configure_cc(c: &mut cc::Build, target: &Target, c_root_dir: &Path, include_dir: &Path) {
     let compiler = c.get_compiler();
-    // FIXME: On Windows AArch64 we currently must use Clang to compile C code
-    let compiler = if target.os == WINDOWS && target.arch == AARCH64 && !compiler.is_like_clang() {
-        let _ = c.compiler("clang");
+    // FIXME: On Windows AArch64 we currently must use clang or clang-cl to compile C code
+    let compiler = if target.os == WINDOWS
+        && target.arch == AARCH64
+        && !compiler.is_like_clang()
+        && !compiler.is_like_clang_cl()
+    {
+        c.compiler("clang");
```

Comment:
Should we start defaulting to clang-cl instead?

Reply:
Perhaps. If users are more commonly compiling for Windows MSVC than Windows GNU, arbitrarily replacing their requested compiler with `clang-cl` whose arguments are compatible with `cl.exe` seems a lot less error-prone than giving them the `clang` interface. Any `cl.exe`-specific argument in their `CFLAGS` (my issue: #2215) and this fails to compile.

Either way, overwriting compilers - which might have been carefully set up via `CC` environment flags etc - seems very painful. As said below, having `ring` compile natively with `cl.exe` would be awesome. At least it's isolated to ARM64 Windows though.

Reply:
At the time, cl.exe didn't have `uint128_t` and probably there were other issues too. Maybe BoringSSL actually made some changes upstream to avoid requiring uint128_t for the ECC code? If so, then we can probably start supporting cl.exe.  Definitely, I would love it, and I think there's already an issue for it.

Reply:
Cool - I'll try to cross-compile with `cl.exe` from my Windows X64 machine to find out and document what specifically is holding it back if at all.

At the same time I'm inquiring to get access to an ARM64 Windows machine soon, to test what happens when natively compiling on that machine.

Reply:
We would point out that uint128_t is not standardised at all either in [C++](https://en.cppreference.com/w/cpp/types/integer.html) or [C](https://en.cppreference.com/w/c/types/integer.html) so sin-binning the compiler for not supporting something that isn't even a part of the language standard seems misguided as this indicates broken code needing fixed.